### PR TITLE
Edge case fixes for log router resource

### PR DESCRIPTION
**Pull request by fenxiong**
### Summary

Fix some edge cases for log router resource missed in previous PR.  This includes:
1. Initialize os and ioutil fields to avoid NPE upon agent restarts.
2. Update applied status back to none when setting resource known status.

### Testing

Added unit tests for the change.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/taskresource/logrouter/logrouter_unix_test.go [hunk 1]
```diff
@@ -260,3 +263,22 @@ func TestCleanupLogRouterResourceError(t *testing.T) {
...
+	logRouterResource := newMockLogRouterResource(LogRouterTypeFluentd, testFluentdOptions, nil, nil)
+	logRouterResource.Initialize(&taskresource.ResourceFields{}, status.TaskRunning, status.TaskRunning)
+
+	assert.NotNil(t, logRouterResource.statusToTransitions)
+	assert.Equal(t, 1, len(logRouterResource.statusToTransitions))
+	assert.NotNil(t, logRouterResource.os)
+	assert.NotNil(t, logRouterResource.ioutil)
+}
+
+func TestSetKnownStatus(t *testing.T) {
```
Comment (yumex93) [inline agent/taskresource/logrouter/logrouter_unix_test.go]: nit: would be better to add a test case to make sure when appliedStatusUnsafe > knownState, the applied status not get reset to none.

Comment (fenxiong) [reply agent/taskresource/logrouter/logrouter_unix_test.go]: added the test

**Comment by fenxiong** [comment]
> did you catch this during release testing?

This was spotted by @yumex93 by eyes :joy: i have manually verified that without the fix there will be NPE, and this PR fixes it.
